checkov/terraform/checks/data/aws/ResourcePolicyDocument.py

```diff
@@ -1,38 +1,21 @@
-from typing import Dict, List, Any
+from __future__ import annotations
 
-from checkov.terraform.checks.data.base_check import BaseDataCheck
-from checkov.common.models.enums import CheckResult, CheckCategories
+from typing import Any, TYPE_CHECKING
 
+from checkov.terraform.checks.data.base_cloudsplaining_data_iam_check import BaseTerraformCloudsplainingDataIAMCheck
 
-class ResourcePolicyDocument(BaseDataCheck):
+if TYPE_CHECKING:
+    from cloudsplaining.scan.policy_document import PolicyDocument
+
+
+class ResourcePolicyDocument(BaseTerraformCloudsplainingDataIAMCheck):
     def __init__(self) -> None:
-        name = 'Ensure IAM policies limit resource access'
+        name = 'Ensure no IAM policies documents allow "*" as a statement\'s resource for restrictable actions'
         id = "CKV_AWS_356"
-        supported_data = ["aws_iam_policy_document"]
-        categories = [CheckCategories.IAM]
-        super().__init__(name=name, id=id, categories=categories, supported_data=supported_data)
-
-    def scan_data_conf(self, conf: Dict[str, List[Any]]) -> CheckResult:
-        """
-            validates iam policy document
-            https://learn.hashicorp.com/terraform/aws/iam-policy
-        :param conf: aws_kms_key configuration
-        :return: <CheckResult>
-        """
-
-        for statement in conf.get("statement", []):
-            if not isinstance(statement, list):
-                statement = [statement]
-            for stmt in statement:
-                if (
-                    isinstance(stmt, dict)
-                    and stmt.get("effect", ["Allow"]) == ["Allow"]
-                    and stmt.get("resources")
-                    and ["*"] == stmt["resources"][0]
-                ):
-                    return CheckResult.FAILED
-
-        return CheckResult.PASSED
+        super().__init__(name=name, id=id)
+
+    def cloudsplaining_analysis(self, policy: PolicyDocument) -> list[str] | list[dict[str, Any]]:
+        return policy.all_allowed_unrestricted_actions
```

Comment:
I'm new to how these are implemented, but so apologies for the question, but where is this list of "allowed unrestricted actions" defined? Is it built-in or configurable?

The reason I ask is that for my use case, when running ECS tasks on ECS Fargate, the task executor needs "ecr:GetAuthorizationToken" on "*". Fargate doesn't seem to function properly when given restricted  resources for that action.

Reply:
it is generated in the `policy_sentry` project https://github.com/salesforce/policy_sentry/blob/master/.github/workflows/update.yml